### ISSUE TITLE
Migrate command should reject deploying if network id does not match

### ISSUE THREAD

**Issue by xavierlepretre** [trufflesuite/truffle#211]
My `truffle.js` contains:

``` javascript
networks: {
  "net42": {
    network_id: 42
  },
  "b9": {
    network_id: 14658
  }
}
```

I have Geth running with `geth --networkid 42 ...`.
Now I run `truffle migrate --network b9`.
And it deployed the contracts. But a look at `Migrations.sol.js` shows that in `Contract.all_networks["14568"]` I have the deployed value in my Geth with network id of 42.
I would have preferred that Truffle complains about the mismatch of network id and fails.

**Comment by tcoulter** [comment]
This should have changed in Truffle 3. Will add to our list to double check.

**Comment by xavierlepretre** [comment]
In v3.4.9, I can still silently deploy to the "wrong" network.

**Comment by gnidan** [comment]
This is done and released in v5.0.0. Let us know if you run into it again! Thanks!
